core/controllers/email_dashboard_test.py

```diff
@@ -152,6 +184,262 @@ def setUp(self):
         self.set_admins(
             [self.SUBMITTER_USERNAME, self.NEW_SUBMITTER_USERNAME])
 
+    def test_handler_with_invalid_num_queries_to_fetch_raises_error_400(self):
+        self.login(self.SUBMITTER_EMAIL)
+
+        response = self.get_json(
+            '/emaildashboarddatahandler',
+            params={'num_queries_to_fetch': '-5'},
```

Comment:
What happens if instead of -5, you pass a random string? or if you give a number above the number of available queries?

Reply:
It doesn't matter if I put a number greater than available queries, it will then return all the queries. This does not fit to this test as this test is for an exception.

I also added a test for random string.